pkg/retry/retry.go

```diff
@@ -77,3 +85,80 @@ func OnError(
 		}
 	}
 }
+
+// Call executes a function repeatedly until it succeeds or the retry strategy
+// indicates that no more retries should be attempted.
```

Comment:
#PUC it seems to me that the "retry strategy" doesn't determine when to stop, that's the responsibility of the "retry function". :thinking: 

```suggestion
// Call executes a function repeatedly, according to the retry strategy until
// it succeeds or returns an `ErrNonRetryable`, which indicates that no more
// retries should be attempted.
```